spec/classes/datadog_agent_spec.rb

```diff
@@ -2177,18 +2177,28 @@
         {
           agent_major_version: 6,
           puppet_run_reports: true,
-          facts_to_tags: ['osfamily', 'facts_array'],
+          facts_to_tags: ['osfamily', 'facts_array', 'facts_hash.actor.first_name', 'looks.like.a.path'],
```

Comment:
What should the output be with something like `facts_hash.path.like.key` and:
```
'facts_hash' => {
  'path.like.key' => 'something',
},
```
? I am not sure what is it that we want in that case.

Reply:
In general, I would say that keys like `path.like.key` shouldn't be used... This PR handles this case only to not break backwards compatibility (in case someone is using weird keys like that), but I don't think we need to support every edge case down that rabbit hole. Otherwise..
```
'facts_hash' => {
  'path.like.key' => {
     "actual.hash" => "I'm evil"
  }
},
```

> Docs say this should be referenced as
Which docs ?

Reply:
>> Docs say this should be referenced as `'facts_hash."path.like.key"'`; we could add a test with that.
>
> Which docs? That syntax doesn't work with the current code.

The Puppet 6.18 [`get` function docs](https://puppet.com/docs/puppet/6.18/function.html#get), just above the one you link in the PR's description (`getvar` calls `get` at some point).

My point with this was that if what we want is to truly replicate `getvar` behavior we should deal with this; it's fine for me if we don't, it doesn't look like a very likely case.